spyderlib/restart_app.py

```diff
@@ -135,6 +144,29 @@ def main():
         time.sleep(0.2)  # Throttling control
 
     env = os.environ.copy()
+
+    # Reset spyder if needed
+    if reset:
+        # Build reset command
+        command_reset = '"{0}" "{1}" {2}'.format(python, spyder, args_reset)
+
+        # Try to reset
+        try:
+            p = subprocess.Popen(command_reset, shell=shell)
+            p.communicate()
+            pid_reset = p.pid
+        except Exception as error:
+            print(command)
+            print(error)

```

Comment:
Using the `traceback` module would be nicer here, since it prints the full traceback as if the Exception wasn't catched. You can use [`traceback.print_exc()`](https://docs.python.org/3.4/library/traceback.html#traceback.print_exc) with no arguments, it prints the right thing at the right place.